networkx/algorithms/clique.py

```diff
@@ -582,10 +583,66 @@ def node_clique_number(G, nodes=None, cliques=None, separate_nodes=False):
 
 
 def number_of_cliques(G, nodes=None, cliques=None):
-    """Returns the number of maximal cliques for each node.
+    """Return the number of maximal cliques each node is part of.
 
-    Returns a single or list depending on input nodes.
+    Output is a single node or list depending on input nodes.
     Optional list of cliques can be input if already computed.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        An undirected graph.
+
+    nodes : list or None, optional (default=None)
+        A list of nodes to return the number of maximal cliques for.
+        If `None`, return the number of maximal cliques for all nodes.
+
+    cliques : list or None, optional (default=None)
+        A precomputed list of maximal cliques to use for the calculation.
+
+    Returns
+    -------
+    int or dict
+        If `nodes` is a single node, return the number of maximal cliques it is
+        part of. If `nodes` is a list, return a dictionary keyed by node to the
+        number of maximal cliques it is part of.
+
+    Raises
+    ------
+    NetworkXNotImplemented
+        If `G` is not undirected.
```

Comment:
Not a blocker, but I slightly prefer avoiding the double-negative if possible!

```suggestion
    NetworkXNotImplemented
        If `G` is directed.
```